scripts/check_sample_grads.py

```diff
@@ -220,12 +223,14 @@ def check_samples():
         mem_lstd_v0_unflat = mem_lstd_v0.reshape(-1, mem_params.shape[-1])
 
         if unrolling_steps > 0:
-            all_mem_grads = jnp.zeros((n_mem, amdp.n_obs, amdp.n_actions, n_mem) +
-                                      mem_params.shape)
+            all_mem_grads = jnp.zeros((n_mem, amdp.observation_space.n, amdp.action_space.n,
+                                       n_mem) + mem_params.shape)
 
             print(f"Calculating memory grads for update {g}")
-            for o, m in tqdm(list(product(list(range(amdp.n_obs)), list(range(n_mem))))):
-                for a, next_m in list(product(list(range(amdp.n_actions)), list(range(n_mem)))):
+            for o, m in tqdm(
```

Comment:
probably not for this PR, but what is all of the `tqdm(list(product(list(range...` for here? Could this pattern be placed in a more descriptive helper?

Reply:
Refactored this - it turns out `product` can take `Iterables` as input - got rid of the `list` calls (except for the `tqdm` call).